analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/OperationProcessors/Invocation.cs

```diff
@@ -33,10 +35,36 @@ internal static class Invocation
                 : context.State;
 
         public static ProgramState Process(SymbolicContext context, IArgumentOperationWrapper argument) =>
-            argument.Parameter is not null      // __arglist is not assigned to a parameter
-            && argument.Parameter.RefKind != Microsoft.CodeAnalysis.RefKind.None
-            && argument.Value.TrackedSymbol() is { } symbol
-                ? context.State.SetSymbolValue(symbol, null)
-                : context.State;
+            ProcessArgument(context.State, argument) ?? context.State;
+
+        private static ProgramState ProcessArgument(ProgramState state, IArgumentOperationWrapper argument) =>
+            argument switch
+            {
+                { Parameter: null } => null, // __arglist is not assigned to a parameter
+                { Parameter.RefKind: not RefKind.None, Value: { } value } when value.TrackedSymbol() is { } symbol =>
+                    // The argument is passed by some kind of reference, so we need to forget all we knew about it.
+                    state.SetSymbolValue(symbol, null),
+                { Parameter: { } parameter } when parameter.GetAttributes() is { Length: > 0 } attributes =>
```

Comment:
We already know it cannot be null from the 1st branch. So we don't need to re-check that
```suggestion
                _ when argument.Parameter.GetAttributes() is { Length: > 0 } attributes =>
```